src/bitcoin-cli.cpp

```diff
@@ -121,14 +121,14 @@ static int AppInitRPC(int argc, char* argv[])
         }
         return EXIT_SUCCESS;
     }
-    if (!fs::is_directory(GetDataDir(false))) {
-        fprintf(stderr, "Error: Specified data directory \"%s\" does not exist.\n", gArgs.GetArg("-datadir", "").c_str());
-        return EXIT_FAILURE;
-    }
     if (!gArgs.ReadConfigFiles(error, true)) {
         fprintf(stderr, "Error reading configuration file: %s\n", error.c_str());
         return EXIT_FAILURE;
     }
+    if (!fs::is_directory(GetDataDir(false))) {
```

Comment:
It seems never fail unless user specify unpermitted path.

Reply:
It fails if the specified path is not a directory. I think that is sensible and was already previous behavior. Or do have a specific change request?

Reply:
If I specify a file, `boost::filesystem::create_directory` throw an error.
If I specify a non-exist path, a new directory is created and no error happens.

I think you just missed second argument; `GetDataDir(false, false)` might be what you want.

Reply:
Sorry for my poor English. Please feel free to ask me if my English is not clear.